src/discussions/posts/post/Post.jsx

```diff
@@ -27,16 +27,30 @@ function Post({
   const dispatch = useDispatch();
   const topic = useSelector(selectTopic(post.topicId));
   const topicContext = useSelector(selectTopicContext(post.topicId));
+  const [isDeleting, showDeleteConfirmation, hideDeleteConfirmation] = useToggle(false);
   const actionHandlers = {
-    [ContentActions.EDIT_CONTENT]: () => history.push({ ...location, pathname: `${location.pathname}/edit` }),
-    // TODO: Add flow to confirm before deleting
-    [ContentActions.DELETE]: () => dispatch(removeThread(post.id)),
+    [ContentActions.EDIT_CONTENT]: () => history.push({
+      ...location,
+      pathname: `${location.pathname}/edit`,
+    }),
+    [ContentActions.DELETE]: showDeleteConfirmation,
     [ContentActions.CLOSE]: () => dispatch(updateExistingThread(post.id, { closed: !post.closed })),
     [ContentActions.PIN]: () => dispatch(updateExistingThread(post.id, { pinned: !post.pinned })),
     [ContentActions.REPORT]: () => dispatch(updateExistingThread(post.id, { flagged: !post.abuseFlagged })),
   };
   return (
-    <div className="d-flex flex-column p-2.5 w-100 mw-100">
+    <div className="d-flex flex-column p-2.5 w-100 mw-100" data-testid={`post-${post.id}`}>
+      <DeleteConfirmation
+        isOpen={isDeleting}
+        title={intl.formatMessage(messages.deletePostTitle)}
+        description={intl.formatMessage(messages.deletePostDescription)}
+        onClose={hideDeleteConfirmation}
+        onDelete={() => {
+          dispatch(removeThread(post.id));
+          history.goBack();
```

Comment:
If we use `history.goBack()`, then there is a discrepency in the user experience based on if the deleted post is the first one the user had opened or if the user comes to this post from some other post.

To provide a uniform experience to the user, I would suggest using this :
```suggestion
          history.push(".");
```

Reply:
I didn't know about this! Thanks!